duckduckgo/macos-browser
# Modularize debug menu and login items logic #3

Task/Issue URL:  https://app.asana.com/0/1199230911884351/1205144339578924/f

## Stack:
- https://github.com/duckduckgo/macos-browser/pull/1378
   - https://github.com/duckduckgo/macos-browser/pull/1389
   - https://github.com/duckduckgo/macos-browser/pull/1397 <- This PR

## Description:

Addresses some feedback in the base PR:
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274649031
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274654723
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274655786
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274661492
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274669463
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274673395
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1274683933

## Testing:

### Test 1: Debug Menu

Test the debug menu and make sure all works well.

1. Reset NetP's status.
2. Remove the system extension and login items.
3. Change your selected server
4. Simulate a failure.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
